### ISSUE TITLE
Remaining cleanup for PyTorch migration

### ISSUE THREAD

**Issue by ShreyaR** [ludwig-ai/ludwig#1374]
- [x] Log statements
- [ ] shape vs. size
- [ ] Rename fc_size to output_size everywhere
- [ ] Replace `module.name` with `module.__get_name()`
- [x] Remove `training`, `mask` arguments from `forward` calls --- not remove mask for sequence, text, time series.
- [ ] Names and parameters of optimizers are consistent with PyTorch naming (E.g. FCStack)
- [ ] Double check defaults (`defaults.py`)
- [ ] Check consistency with asserts / checks for types and relative errors (RuntimeError vs ValueError and other errors)
- [ ] Clean up: Check that self.overwrite_defaults() can be removed in *_features' constructors as overwrite_defaults() already called by the base class's constructor.
- [ ] Clean up "Copyright (c) 2019 Uber Technologies, Inc." headers?
- [ ] Check pythonic asserts in encoders unit tests. (Justin)
- [ ] Add @overload annotations for subclassed overridden methods.
